Source/WebCore/platform/graphics/displaylists/DisplayListItems.h

```diff
@@ -1207,81 +914,30 @@ class FillRectWithColor {
 
     WEBCORE_EXPORT void apply(GraphicsContext&) const;
 
-    template<class Encoder> void encode(Encoder&) const;
-    template<class Decoder> static std::optional<FillRectWithColor> decode(Decoder&);
-
 private:
     FloatRect m_rect;
     Color m_color;
 };
 
-template<class Encoder>
-void FillRectWithColor::encode(Encoder& encoder) const
-{
-    encoder << m_rect;
-    encoder << m_color;
-}
-
-template<class Decoder>
-std::optional<FillRectWithColor> FillRectWithColor::decode(Decoder& decoder)
-{
-    std::optional<FloatRect> rect;
-    decoder >> rect;
-    if (!rect)
-        return std::nullopt;
-
-    std::optional<Color> color;
-    decoder >> color;
-    if (!color)
-        return std::nullopt;
-
-    return {{ *rect, *color }};
-}
-
 class FillRectWithGradient {
 public:
     static constexpr ItemType itemType = ItemType::FillRectWithGradient;
     static constexpr bool isInlineItem = false;
     static constexpr bool isDrawingItem = true;
 
     WEBCORE_EXPORT FillRectWithGradient(const FloatRect&, Gradient&);
```

Comment:
We might be able to remove this constructor.

Reply:
It's currently in use by our DisplayListRecorder